lib/recipes-data/src/lib/transform.ts

```diff
@@ -1,84 +1,109 @@
-import { FeaturedImageWidth, ImageDpr, PreviewImageWidth } from './config';
-import type { Contributor, RecipeImage } from './models';
-import { extractCropDataFromGuimUrl } from './utils';
+import type {Sponsorship} from "@guardian/content-api-models/v1/sponsorship";
+import {SponsorshipType} from "@guardian/content-api-models/v1/sponsorshipType";
+import {FeaturedImageWidth, ImageDpr, PreviewImageWidth} from './config';
+import type {Contributor, RecipeImage} from './models';
+import {extractCropDataFromGuimUrl} from './utils';
+
+export type RecipeTransformationFunction = (recipeData: Record<string, unknown>) => Record<string, unknown>;
 
 const getFastlyUrl = (
-	imageId: string,
-	cropId: string,
-	dpr: number,
-	desiredWidth: number,
-	originalWidth: number,
+  imageId: string,
+  cropId: string,
+  dpr: number,
+  desiredWidth: number,
+  originalWidth: number,
   extension: string
 ) => `https://i.guim.co.uk/img/media/${imageId}/${cropId}/master/${originalWidth}.${extension}?width=${desiredWidth}&dpr=${dpr}&s=none`;
 
 export const replaceFastlyUrl = (
-	recipeId: string,
-	image: RecipeImage,
-	desiredWidth: number,
-	dpr: number,
+  recipeId: string,
+  image: RecipeImage,
+  desiredWidth: number,
+  dpr: number,
 ): RecipeImage => {
   const cropData = extractCropDataFromGuimUrl(image.url);
 
-	if (!cropData) {
-		console.warn(
-			`Error adding fastly URL to recipe ${recipeId} - no crop data found for image with url ${image.url}.`,
-		);
-		return image;
-	}
+  if (!cropData) {
+    console.warn(
+      `Error adding fastly URL to recipe ${recipeId} - no crop data found for image with url ${image.url}.`,
+    );
+    return image;
+  }
 
-	const { mediaId, cropId, width, extension } = cropData;
+  const {mediaId, cropId, width, extension} = cropData;
 
-	return {
-		...image,
-		url: getFastlyUrl(mediaId, cropId, dpr, desiredWidth, width, extension),
-	};
+  return {
+    ...image,
+    url: getFastlyUrl(mediaId, cropId, dpr, desiredWidth, width, extension),
+  };
 };
 
 export type RecipeWithImageData = {
-	id: string;
-	featuredImage: RecipeImage | string; // the latter is an old image format that appears in our test fixtures
-	previewImage?: RecipeImage | string;
+  id: string;
+  featuredImage: RecipeImage | string; // the latter is an old image format that appears in our test fixtures
+  previewImage?: RecipeImage | string;
+};
+
+const getSponsorshipTypeNames: Record<number, string> = {
+  [SponsorshipType.SPONSORED]: "Sponsored",
+  [SponsorshipType.FOUNDATION]: "Foundation",
+  [SponsorshipType.PAID_CONTENT]: "Paid content"
 };
 
+export const addSponsorsTransform: (sponsors: Sponsorship[]) => RecipeTransformationFunction = sponsors => {
+  return (recipeData) => ({
+    ...recipeData,
+    sponsors: sponsors.length === 0 ? undefined : sponsors.map(sponsor => {
+      return {
+        ...sponsor,
+        sponsorshipType: getSponsorshipTypeNames[sponsor.sponsorshipType],
```

Comment:
We should deal with an "unknown" value here - `sponsorshipType` is a mandatory field in Thrift, so we should put something like `getSponsorshipTypeNames[sponsor.sponsorshipType] ?? "Unknown"` in case having an "undefined" value breaks the client json parsing

Reply:
@Divs-B if you make the change above then this won't be necessary I hope

Reply:
addressed in above change. thanks @fredex42